### PR TITLE
Update K8s plugin config docs

### LINKED ISSUE #5065: [Docs] Update Spark config documentation
### Description

The following issues have been detected in the [K8s plugins configuration page](https://docs.flyte.org/en/latest/deployment/plugins/k8s/index.html):

1. No reference to how to configure the Spark plugin for a GCP deployment
2. It asks to create four separate files for the Spark ServiceAccount, Role, RoleBinding and even the `deault` Service Account. The problem is that:
  - This is not necessary as all four blocks can be wrapped inside the `templates` section. It has been tested and works
  - The YAML manifests use Helm variable references that won't be applied when submitted as standalone YAML

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by davidmirror-ops**
## Tracking issue
Closes #5065

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

**Comment by neverett** [comment]
@davidmirror-ops OK if i review this now or would you like me to wait until it's marked as ready for review and not draft?

**Comment by davidmirror-ops** [comment]
@neverett ready for review :)

File: docs/deployment/plugins/k8s/index.rst [hunk 1]
```diff
@@ -5,468 +5,79 @@ Configure Kubernetes Plugins
...
-                  namespace: "{{ namespace }}"
-                roleRef:
-                  apiGroup: rbac.authorization.k8s.io
-                  kind: Role
-                  name: spark-role
-                subjects:
-                  - kind: ServiceAccount
-                    name: spark
-                    namespace: "{{ namespace }}"
+This guide help you configure the Flyte plugins that provision resources on Kubernetes.
```
Comment (neverett) [inline docs/deployment/plugins/k8s/index.rst:8]:
```suggestion
This guide will help you configure the Flyte plugins that provision resources on Kubernetes.
```

File: docs/deployment/plugins/k8s/index.rst [hunk 2]
```diff
@@ -5,468 +5,79 @@ Configure Kubernetes Plugins
...
+            metadata:
+              annotations:
+                yunikorn.apache.org/task-group-name: ""
+                yunikorn.apache.org/task-groups: ""
+                yunikorn.apache.org/schedulingPolicyParameters: ""
+              
+              
+    See :ref:`deployment-configuration-general` for more information about Pod templates in Flyte.
+    You can set the scheduler name in the Pod template passed to the ``@task`` decorator. However, to prevent the
+    two different schedulers from competing for resources, it is recommended to set the scheduler name in the pod template
```
Comment (neverett) [inline docs/deployment/plugins/k8s/index.rst:62]:
```suggestion
    two different schedulers from competing for resources, we recommend setting the scheduler name in the pod template
```

File: docs/deployment/plugins/k8s/index.rst [hunk 3]
```diff
@@ -693,271 +309,666 @@ Specify plugin configuration
...
+                            - spark.hadoop.fs.AbstractFileSystem.s3.impl: "org.apache.hadoop.fs.s3a.S3A"
+                            - spark.hadoop.fs.s3a.impl: "org.apache.hadoop.fs.s3a.S3AFileSystem"
+                            - spark.hadoop.fs.AbstractFileSystem.s3a.impl: "org.apache.hadoop.fs.s3a.S3A"
+                            - spark.network.timeout: 600s
+                            - spark.executorEnv.KUBERNETES_REQUEST_TIMEOUT: 100000
+                            - spark.executor.heartbeatInterval: 60s
+                
+              2. (Optional) The Spark operator supports Kubernetes ResourceQuota enforcement. If you plan to use it, 
+                 set `per-Task resource requests <https://docs.flyte.org/en/latest/user_guide/productionizing/customizing_task_resources.html#customizing-task-resources>`__ that fit into the quota for each project-namespace. A Task without resource requests
+                 or limits will be rejected by the K8s scheduler as described `in the docs <https://kubernetes.io/docs/concepts/policy/resource-quotas/>`__.
```
Comment (neverett) [inline docs/deployment/plugins/k8s/index.rst:439]:
```suggestion
                 or limits will be rejected by the K8s scheduler as described `in the Kubernetes docs <https://kubernetes.io/docs/concepts/policy/resource-quotas/>`__.
```

**Review by neverett — DISMISSED** [review]
A few small suggestions, otherwise LGTM!

**Review by neverett — APPROVED** [review]

**Pull request merged**